sockeye/inference.py

```diff
@@ -1193,7 +1196,10 @@ def _get_inference_input(self,
                 raw_constraints[j] = [data_io.tokens2ids(phrase, self.vocab_target) for phrase in
                                       trans_input.constraints]
 
-        return source, bucket_key, raw_constraints
+        # max output lengths extended to shape (batch_size * beam_size,)
+        max_out_lengths_extended = mx.nd.repeat(mx.nd.array(max_output_lengths, ctx=self.context, dtype='int32'), self.beam_size)
```

Comment:
maybe just return the NDArray and do the actual `mx.nd.repeat` in `_beam_search` because it is only needed due to the fact that we have a beam. If we were to add something else, say greedy search, we could use the same method to create the input, while internally beam search would need to repeat the lengths.